src/ui/cmake/Dependencies.cmake

```diff
@@ -52,19 +52,20 @@ FetchContent_Declare(
     GIT_TAG 4.2.2
 )
 
-if (EMSCRIPTEN)
-    FetchContent_MakeAvailable(imgui implot imgui-node-editor yaml-cpp stb opencmw-cpp plf_colony function2)
-else () # native build
-    FetchContent_Declare(
-             sdl2
-             GIT_REPOSITORY "https://github.com/libsdl-org/SDL"
-             GIT_TAG        release-2.28.2
-    )
-    FetchContent_MakeAvailable(imgui implot imgui-node-editor yaml-cpp plf_colony stb opencmw-cpp function2)
-    find_package(SDL2 REQUIRED)
-    find_package(OpenGL REQUIRED COMPONENTS OpenGL)
-    # target_link_libraries(SDL2 PUBLIC OpenGL::GL )
-    # target_include_directories(SDL2 PUBLIC ${sdl2_SOURCE_DIR}/include)
+FetchContent_MakeAvailable(imgui implot imgui-node-editor yaml-cpp stb opencmw-cpp plf_colony function2)
+
+if (NOT EMSCRIPTEN)
+FetchContent_Declare(
+        sdl2
+        OVERRIDE_FIND_PACKAGE
+        GIT_REPOSITORY "https://github.com/libsdl-org/SDL"
+        GIT_TAG        release-2.28.2
+)
+FetchContent_MakeAvailable(sdl2)
+
+find_package(SDL2 REQUIRED)
+find_package(OpenGL REQUIRED COMPONENTS OpenGL)
+
```

Comment:
"Projects should aim to declare the details of all dependencies they might use before they call FetchContent_MakeAvailable() for any of them." https://cmake.org/cmake/help/latest/module/FetchContent.html

It might not be relevant in this case, but that was the idea of the original version.